### PR TITLE
Fix staticcheck report

### PR DESCRIPTION
**Pull request by kozlovic**
Removed `os.Exit()` from function assigned to `flag.Usage`.

### CODE REVIEW

**Comment by kozlovic** [comment]
With this change, server still exit on wrong usage. Here is an example:
```
Ivans-MacBook-Pro:gnatsd ivan$ gnatsd -x
flag provided but not defined: -x

Usage: gnatsd [options]

Server Options:
    -a, --addr <host>                Bind to host address (default: 0.0.0.0)
[...]
    -h, --help                       Show this message
    -v, --version                    Show version
        --help_tls                   TLS help

Ivans-MacBook-Pro:gnatsd ivan$ 
```

File: main.go [hunk 1]
```diff
@@ -62,7 +62,6 @@ Common Options:
 // usage will print out the flag options for the server.
 func usage() {
 	fmt.Printf("%s\n", usageStr)
-	os.Exit(0)
```
Comment (derekcollison) [inline main.go]: This new behavior that it is not required?

Comment (kozlovic) [inline main.go]: actually that fails with `gnatsd help` so don't merge!

Comment (kozlovic) [inline main.go]: I will update the PR, but we call `flag.Parse()` which is `CommandLine.Parse(os.Args[1:])`. and `CommandLine` is defined as `var CommandLine = NewFlagSet(os.Args[0], ExitOnError)`. Which means that if there is a parsing error, `flag` will call `os.Exit(2)`. Notice that it does not return `0`.
Two ways of fixing:
1- I keep `usage()` as it is now and do `flag.Usage = func() { fmt.Printf("%s\n", usageStr) }
2- I silence this staticcheck warning
Which one do you prefer?

**Comment by kozlovic** [comment]
@derekcollison I have updated PR by leaving the `os.Exit(0)` in our `usage` function since we call this one when one does: `gnatsd help`, but I have used an anonymous function to simply print the usage banner for `flag.Usage()`.

File: main.go [hunk 2]
```diff
@@ -119,7 +119,9 @@ func main() {
 	flag.StringVar(&opts.TLSKey, "tlskey", "", "Private key for server certificate.")
 	flag.StringVar(&opts.TLSCaCert, "tlscacert", "", "Client certificate CA for verification.")
 
-	flag.Usage = usage
+	flag.Usage = func() {
+		fmt.Printf("%s\n", usageStr)
```
Comment (derekcollison) [inline main.go]: Did we remove usage func or is it used elsewhere?

Comment (kozlovic) [inline main.go]: I did a `git push --force`. The usage function is still there (top of the file) and I have put back the `os.Exit(0)` call. Again, we call `usage()` directly if user provides `help` command line param.

**Pull request merged**